src/components/navbar-header-panels/side-menu-panel.component.tsx

```diff
@@ -1,15 +1,29 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { ExtensionSlot } from '@openmrs/esm-framework';
 import { SideNav } from 'carbon-components-react/es/components/UIShell';
 import { SideNavProps } from 'carbon-components-react';
 import styles from './side-menu-panel.component.scss';
 
-interface SideMenuPanelProps extends SideNavProps {}
+interface SideMenuPanelProps extends SideNavProps {
+  hidePanel: Function;
+}
+
+const SideMenuPanel: React.FC<SideMenuPanelProps> = ({ expanded, hidePanel }) => {
+  const menuRef = React.useRef(null);
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (menuRef.current && !menuRef.current.contains(event.target)) {
+        hidePanel();
+      }
+    };
+
+    document.addEventListener('click', handleClickOutside);
+    return () => document.removeEventListener('click', handleClickOutside);
+  }, [menuRef, hidePanel]);
```

Comment:
I think `menuRef` is stable, but `menuRef.current` is not.

Reply:
should i just do `menuRef?.current?blahblah` just in case? or like is it not stable as an api

Reply:
I mean what you care about in case of dependencies is `menuRef.current`. If that reference is different you need to be active - because you need to move your handler.

Reply:
got it - ill add it to the deps array

Reply:
https://github.com/openmrs/openmrs-esm-primary-navigation/pull/80/commits/0aeb8fca9f637080a33347711cbee6051f2b5352